docs/source/user/debugging.rst

```diff
@@ -0,0 +1,166 @@
+Debugging Numba CUDA Programs with Visual Studio Code and CUDA GDB
+==================================================================
+
+Introduction
+------------
+
+With the release of the CUDA Toolkit (CTK) 13.1, CUDA GDB now includes beta support for debugging Numba CUDA programs on Linux. CUDA GDB is included in the CTK and is the backend debugger for the Nsight Visual Studio Code Edition extension to support debugging in Microsoft Visual Studio Code (``VSCode``).
+
+Features included in this release:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* Linux support
+* Debugging of Numba CUDA programs with CUDA GDB using ``VSCode``.
+* Debugging of Numba CUDA programs with CUDA GDB from the command line.
+* Variable inspection and modification.
+* Execution control (continue, step over, step into, step out, restart, stop).
+* Setting breakpoints in the VSCode GUI.
+* Setting breakpoints programmatically in Numba CUDA code by inserting a ``breakpoint()`` call.
+* Formatting arrays in a more human readable format.
+* Basic support for polymorphic variables.
+
+These directions are for debugging with CUDA GDB using ``VSCode``, but CLI debugging with CUDA GDB is also supported. A more detailed description of debugging with VSCode in general can be found here: https://code.visualstudio.com/docs/editor/debugging.
+
+Installation and Environment Setup
+----------------------------------
+
+To begin you’ll need an installation of CUDA Toolkit 13.1 and corresponding CUDA Driver for Linux. The installer for Linux can be downloaded from https://developer.nvidia.com/cuda-downloads. Follow the instructions provided on the download page to install it in ``/usr/local/cuda-13.1`` and create a symlink to it called ``/usr/local/cuda`` (which is handled by the install script).
+
+You will also need a working Numba CUDA development environment. The documentation for Numba CUDA including the installation instructions can be found at: https://nvidia.github.io/numba-cuda. It is highly recommended that you use Anaconda, venv, or another Python virtual environment. The examples in this documentation are based on an Anaconda environment. In particular, make sure the required packages are installed in the virtual environment you wish to debug in.
+
+The following commands can be used to create a Numba CUDA development environment using Anaconda. This example creates a virtual environment called ``numba-cuda-debug`` using Python 3.12.
+
+.. code-block:: bash
+
+    conda create --name numba-cuda-debug python=3.12
+    conda activate numba-cuda-debug
+    pip install cuda-python numba numba-cuda
```

Comment:
We already have installation instructions in the documentation: https://nvidia.github.io/numba-cuda/user/installation.html

This seems to be partially duplicating them, so I think it's probably be more concise and straightforward to omit this too?